src/vulkan/device.cc

```diff
@@ -511,13 +521,33 @@ Result Device::Initialize(const std::vector<Feature>& required_features,
     r = CreateDevice(required_features, required_extensions);
     if (!r.IsSuccess())
       return r;
-  }
+  } else {
+    if (!AreAllRequiredFeaturesSupported(physical_device_, required_features)) {
```

Comment:
The CTS can give us a bunch of this information, see https://github.com/KhronosGroup/VK-GL-CTS/blob/master/external/vulkancts/modules/vulkan/vktTestCase.hpp. Would it be easier to just pass in the features, extensions and queue instead of re-querying? In that case, do we need the physical device?